src/pokedex_screen.c

```diff
@@ -242,21 +240,21 @@ void sub_810250C(void)
     ScanlineEffect_Stop();
     ResetBgsAndClearDma3BusyFlags(TRUE);
     InitBgsFromTemplates(0, gUnknown_8451EBC, NELEMS(gUnknown_8451EBC));
-    SetBgTilemapBuffer(3, Alloc(BG_SCREEN_SIZE));
-    SetBgTilemapBuffer(2, Alloc(BG_SCREEN_SIZE));
-    SetBgTilemapBuffer(1, Alloc(BG_SCREEN_SIZE));
-    SetBgTilemapBuffer(0, Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(3, (u16*)Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(2, (u16*)Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(1, (u16*)Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(0, (u16*)Alloc(BG_SCREEN_SIZE));
     if (natDex)
-        DecompressAndLoadBgGfxUsingHeap(3, gUnknown_84403AC, BG_SCREEN_SIZE, 0, 0);
+        DecompressAndLoadBgGfxUsingHeap(3, (void*)gUnknown_84403AC, BG_SCREEN_SIZE, 0, 0);
     else
-        DecompressAndLoadBgGfxUsingHeap(3, gUnknown_8440274, BG_SCREEN_SIZE, 0, 0);
+        DecompressAndLoadBgGfxUsingHeap(3, (void*)gUnknown_8440274, BG_SCREEN_SIZE, 0, 0);
```

Comment:
Is there a reason for these casts?

Reply:
Yes, because otherwise gcc throws warnings